hazelcast-client/src/main/java/com/hazelcast/client/impl/HazelcastClientInstanceImpl.java

```diff
@@ -242,6 +244,8 @@ public HazelcastClientInstanceImpl(ClientConfig config,
         lockReferenceIdGenerator = new ClientLockReferenceIdGenerator();
         nearCacheManager = clientExtension.createNearCacheManager();
         clientExceptionFactory = initClientExceptionFactory();
+
+        statistics = new Statistics(this);
```

Comment:
I initially had the explicit parameters passed but it came out to be five of them (ClientExecutionService, ClusterService, LifeCycleService, NearCacheManager, ClientProperties) which I thought would be cleaner to pass "this" instead of five parameters. Partial construction will not be a problem for this case but I understand the concern. In addition, "this" is passed HazelcastClientInstanceImpl as a parameter to a number of objects. Do you think it is still better to pass five parameters instead of "this"?